src/map/lua/lua_treasure_pool.cpp

```diff
@@ -0,0 +1,190 @@
+{
+    if (const auto PChar = dynamic_cast<CCharEntity*>(PEntity))
+    {
+        m_PLuaTreasurePool->DelMember(PChar);
+    }
+}
+
+/************************************************************************
+ *  Function: update()
+ *  Purpose : Force refreshes the treasure pool content for a PC
+ *  Example : pool:update(target)
+ *  Notes   :
+ ************************************************************************/
+void CLuaTreasurePool::update(CBaseEntity* PEntity) const
+{
+    if (const auto PChar = dynamic_cast<CCharEntity*>(PEntity))
+    {
+        m_PLuaTreasurePool->UpdatePool(PChar);
+    }
+}
+
+/************************************************************************
+ *  Function: addItem()
+ *  Purpose : Adds an item to the treasure pool
+ *  Example : pool:addItem(xi.item.RIDILL)
+ *  Notes   : The entity can be either a NPC, a mob or nullptr.
+ ************************************************************************/
+auto CLuaTreasurePool::addItem(const uint16 ItemID, CBaseEntity* PEntity) const -> uint8
+{
+    return m_PLuaTreasurePool->AddItem(ItemID, PEntity);
+}
+
+/************************************************************************
+ *  Function: memberCount()
+ *  Purpose : Returns the number of members in the treasure pool
+ *  Example : pool:memberCount()
+ *  Notes   :
+ ************************************************************************/
+auto CLuaTreasurePool::memberCount() const -> size_t
+{
+    return m_PLuaTreasurePool->MemberCount();
+}
+
+/************************************************************************
+ *  Function: getMembers()
+ *  Purpose : Returns all members present in the pool.
+ *  Example : pool:getMembers()
+ *  Notes   :
+ ************************************************************************/
+auto CLuaTreasurePool::getMembers() const -> sol::table
+{
+    sol::table result = lua.create_table();
+    int        index  = 1;
+
+    for (const auto& member : m_PLuaTreasurePool->GetMembers())
+    {
+        result[index++] = member;
+    }
+
+    return result;
+}
+
+/************************************************************************
+ *  Function: getItems()
+ *  Purpose : Returns all items present in the pool, along with their lotters
+ *  Example : pool:getItems()
+ *  Notes   : ID of 0 means no item present in the given slot.
+ *  {
+ *    { ID = 1, SlotID = 0, TimeStamp = 0, Lotters = { { member = {...}, lot = 0 } } },
+ *    [...]
+ *  }
+ ************************************************************************/
+auto CLuaTreasurePool::getItems() const -> sol::table
+{
+    sol::table result = lua.create_table();
+    int        index  = 1;
+
+    for (const auto& item : m_PLuaTreasurePool->GetItems())
+    {
+        sol::table itemRow   = lua.create_table();
+        itemRow["ID"]        = item.ID;
+        itemRow["SlotID"]    = item.SlotID;
+        itemRow["TimeStamp"] = item.TimeStamp.time_since_epoch().count();
```

Comment:
Off the top of my head, I don't think we often have `Capitalized` keys when we build Lua tables